client/src/annoMatrix/query.ts

```diff
@@ -1,22 +1,51 @@
 import sha1 from "sha1";
 import { _dubEncURIComp } from "./fetchHelpers";
+import { Field } from "../common/types/entities";
 
 /**
  * Query utilities, mostly for debugging support and validation.
  */
 
+export type ComplexQuery = SummarizeQuery | WhereQuery;
+
+export type Query = string | ComplexQuery | (string | ComplexQuery)[];
+
+interface SummarizeQuery {
+  summarize: SummarizeQueryTerm;
+}
+
+interface SummarizeQueryTerm {
+  column: string;
+  field: string;
+  method?: string;
```

Comment:
method should not be optional AFAIK

Reply:
Updated:

* [query.ts::SummarizeQueryTerm](https://github.com/chanzuckerberg/cellxgene/blob/mim/annoMatrix/client/src/annoMatrix/query.ts#L20)
* [whereCache.test.ts#L26](https://github.com/chanzuckerberg/cellxgene/blob/mim/annoMatrix/client/__tests__/util/annoMatrix/whereCache.test.ts#L26)